pkg/smartcontract/rpcbinding/binding.go

```diff
@@ -675,36 +674,31 @@ func scTemplateToRPC(cfg binding.Config, ctr ContractTmpl, imports map[string]st
 		imports["errors"] = struct{}{}
 	}
 	for _, abiEvent := range cfg.Manifest.ABI.Events {
+		eBindingName := ToEventBindingName(abiEvent.Name)
 		eTmp := CustomEventTemplate{
-			// TODO: proper event name is better to be set right into config binding in normal form.
-			Name:         toPascalCase(abiEvent.Name),
+			Name:         eBindingName,
 			ManifestName: abiEvent.Name,
 		}
-		var varnames = make(map[string]bool)
 		for i := range abiEvent.Parameters {
-			name := abiEvent.Parameters[i].Name
-			fullPName := abiEvent.Name + "." + name
+			pBindingName := ToParameterBindingName(abiEvent.Parameters[i].Name)
+			fullPName := eBindingName + "." + pBindingName
 			typeStr, pkg := scTypeConverter(fullPName, abiEvent.Parameters[i].Type, &cfg)
```

Comment:
Suppose we have a new NEP with extended type data. Will this scheme still work with compilers pushing random things into names?

Reply:
> compilers pushing random things into names

It's just *our* compiler that pushes prettified names into config *that may be used only by our compiler*. The parameter names must be prettified to generate proper go code (prettification includes capitalization, removing spaces and all not-a-go-name-ish garbage that user may put into event/parameter name. This step is needed anyway and, as discussed in https://github.com/nspcc-dev/neo-go/pull/3008#discussion_r1191228640, the config should contain already prettified names, not just the user's input.

> Suppose we have a new NEP with extended type data

This scheme will work with the new NEP, and I don't expect that this part of code will be removed or rewrited, because notification names can still contain any garbage as far as notification parameter names, and the new NEP doesn't solve this problem. I.e. we need to prettify them anyway.

Reply:
Yes, we do! Here's the test: `TestAssistedRPCBindings`, check the `checkBinding(filepath.Join("testdata", "notifications"), true, "_guessed")` and the resulting output located in the `neo-go/cli/smartcontract/testdata/notifications/rpcbindings_guessed.out`. We also have the same result if no `--guess-eventtypes` is used and no additional information is specified in the contract config. (see the `! complicated name %$#` notification in the contract config)